tools/filelife.py

```diff
@@ -109,12 +151,21 @@
     if args.ebpf:
         exit()
 
+if BPF.kernel_struct_has_field(b'renamedata', b'old_mnt_userns') == 1:
+    bpf_text = bpf_text.replace('TRACE_CREATE_FUNC', trace_create_text_new)
+    bpf_text = bpf_text.replace('TRACE_UNLINK_FUNC', trace_unlink_text_new)
+else:
+    bpf_text = bpf_text.replace('TRACE_CREATE_FUNC', trace_create_text_old)
+    bpf_text = bpf_text.replace('TRACE_UNLINK_FUNC', trace_unlink_text_old)
+
 # initialize BPF
 b = BPF(text=bpf_text)
 b.attach_kprobe(event="vfs_create", fn_name="trace_create")
+# newer kernels may don't fire vfs_create, call vfs_open instead:
+b.attach_kprobe(event="vfs_open", fn_name="trace_open")
```

Comment:
Maybe we should also trace `vfs_tmpfile` ?

Reply:
When I wrote this code, I also thought about vfs_tmpfile 🤝, thanks

Reply:
I check `$ man 2 open`

```
O_TMPFILE (since Linux 3.11) Create an unnamed temporary regular file. Anything written to the resulting file will be lost when the last file descriptor is closed, unless the file is given a name.
```
vfs_tmpfile DON'T have dentry struct, thus, i think we don't need it.